content/_components/deepl/index.md

```diff
@@ -0,0 +1,124 @@
+description: The DeepL Component is specifically crafted to interact with the DeepL API.
+icon: deepl.png
+icontext: DeepL component
+category: deepl
+updatedDate: 2024-05-08
+ComponentVersion: 1.0.0
+---
+
+## Description
+
+The DeepL Component is specifically crafted to interact with the [DeepL API](https://www.deepl.com/docs-api).
+
+The current version of the component has been tested with API version `v2`.
+
+## Authentication
+To utilize the DeepL Component, you must provide the following credentials:
+
+- **Authentication Key** (string, required) - This key is essential for accessing the API. You can easily locate your unique key within your [account settings](https://www.deepl.com/account/summary).
+
+## Triggers
+
+This component has no trigger functions. This means it will not be accessible to
+select as a first component during the integration flow design.
+
+
+## Actions
+
+  * [**Make Raw Request**](#make-raw-request).
+  * [**Translate Document**](#translate-document).
+  * [**Translate Text**](#translate-text).
+
+## Make Raw Request
+Enables you to run your own custom requests directly through the DeepL REST API.
+
+### Configuration Fields
+- **Don't Throw Error on 404 Response** - (optional, boolean): This setting configures the handling of 404 HTTP responses as non-errors. By default, it is set to `false`.
+
+### Input Metadata
+- **URL** - (string, required): The specific path of the resource, which is appended to the base URL `https://api(-free).deepl.com/v2/`.
+- **Method** - (string, required): Determines the HTTP method for the request.
+- **Request Body** - (object, optional): Contains the content for the request.
+
+### Output Metadata
+- **Status Code** - (number, required): Indicates the HTTP response status code.
+- **HTTP Headers** - (object, required): Showcases the HTTP headers of the response.
+- **Response Body** - (object, optional): Represents the body of the HTTP response.
+
+## Translate Document
+This action enables you to translate whole documents and supports the following file types and extensions:
+
+- `docx` - Microsoft Word Document.
+- `pptx` - Microsoft PowerPoint Document.
+- `xlsx` - Microsoft Excel Document.
+- `pdf` - Portable Document Format.
+- `htm` / `html` - HTML Document.
+- `txt` - Plain Text Document.
+- `xlf`/ `xliff` - XLIFF Document, version 2.1.
+
+> **Please note** that with every submitted document of type `.pptx`, `.docx`, `.xlsx`, or `.pdf`, you are billed a minimum of 50,000 characters with the DeepL API plan, no matter how many characters are included in the document.
+    
+### Configuration Fields
+- **Target language** - (dropdown, required): The language into which the text should be translated.
+- **Emit behavior** (dropdown, required) - By default, we need to wait until DeepL translates the file after it was sent to the server periodically checking the status, here you can select when the component should emit a message. Options currently available:
```

Comment:
**Emit behaviour** (dropdown, required) - By default, we need to wait until DeepL translates the file after it was sent to the server periodically checking the status, here you can select when the component should emit a message. Options currently available: